atlp-rwanda/e-commerce-furebo-32-bn
# Configuring Codeclimate And Adding Test coverage

What does this PR do?
This PR integrates test coverage reporting into the project using CodeClimate and adds a coverage badge to the README file to display the current test coverage status.

Describe your changes
1.Added Coverage Reporting:

Installed the codeclimate-test-reporter package.
Configured Jest to generate lcov coverage reports.
Updated CircleCI configuration to run tests, generate coverage reports, and upload them to CodeClimate.
Configured Coverage Thresholds:

2.Enforced a minimum test coverage threshold in CodeClimate settings.
Added Coverage Badge to README:

3.Included a test coverage badge at the top of the README.md file.

PR Type
What kind of change does this PR introduce?

[ ] Bugfix
[x] Feature
[ ] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[x] Tests
[ ] Other

How should this be manually tested?
1.Check Test Coverage Reporting:

Push a commit to trigger the CircleCI build.
Verify that tests run and the coverage report is generated and uploaded to CodeClimate.

2.Check Coverage Badge:

Ensure the coverage badge in the README.md file reflects the current test coverage percentage.

What are the relevant pivotal tracker stories?
[Pivotal Tracker Link](https://www.pivotaltracker.com/n/projects/2703465/stories/187640957)
